alchemy/src/cloudflare/vite-site.ts

```diff
@@ -82,12 +82,25 @@ export async function ViteSite<B extends Bindings>(
     });
 
     const staticAssets = await Assets("assets", {
-      path: props.assets ?? "./dist",
+      path:
+        props.assets ??
+        (await (async () => {
+          try {
+            await fs.access(path.join("dist", "client", "index.html"));
+            return path.join(".", "dist", "client");
+          } catch {
+            return path.join(".", "dist");
+          }
+        })()),
     });
-
     const worker = await Worker("worker", {
       name: props.name ?? id,
       entrypoint: props.main,
+      assets: {
+        html_handling: "auto-trailing-slash",
+        not_found_handling: "single-page-application",
+        run_worker_first: false,
+      },
       script: props.main
         ? undefined
         : `
```

Comment:
do we want to do return new Response("Not Found2", { status: 404 }); here instead too?

Reply:
Good catch! Yes, we should do that.